src/UseCases/AddComment/AddCommentUseCase.php

```diff
@@ -63,9 +72,18 @@ private function requestIsAllowed( AddCommentRequest $addCommentRequest ): bool
 	private function newCommentFromRequest( AddCommentRequest $request ): DonationComment {
 		return new DonationComment(
 			$request->getCommentText(),
-			$request->isPublic(),
+			$this->commentCanBePublic( $request ),
 			$request->getAuthorDisplayName()
 		);
 	}
 
+	private function commentCanBePublic( AddCommentRequest $request ): bool {
+		return $request->isPublic()
+			&& $this->commentTextPassesValidation( $request->getCommentText() );
```

Comment:
Works for me. I suggest to create a ticket for that under New Stuff. One question I have for it is: does the backend need to be modified in any way to match such a change?